include/spirv-tools/optimizer.hpp

```diff
@@ -738,9 +738,10 @@ Optimizer::PassToken CreateCombineAccessChainsPass();
 // |input_length_enable| controls instrumentation of runtime descriptor array
 // references, and |input_init_enable| controls instrumentation of descriptor
 // initialization checking, both of which require input buffer support.
+// |version| specifies the buffer record format.
 Optimizer::PassToken CreateInstBindlessCheckPass(
     uint32_t desc_set, uint32_t shader_id, bool input_length_enable = false,
-    bool input_init_enable = false);
+    bool input_init_enable = false, uint32_t version = 1);
```

Comment:
If version 1 is deprecated, why do we make that the default?  This will encourage people to generate the deprecated version.  We should default to the latest version if we default at all.

Reply:
This entry point is only used for testing, which is still generally done with version 1 since that is what the validation layer is still using. I did add one test using version 2.

Once the layer switches to 2, then I will update all the tests to 2. Maybe I am slightly premature in deprecating version 1, but I thought everything would be clearer to you if you understood that version 1 was going away eventually.